### PR TITLE
Add integration for exim

### PR DESCRIPTION
**Pull request by JeanFred**
### What does this PR do?

Add a new integration for [exim](https://www.exim.org)

### Motivation

I am an exim end-user at $dayJob and want to monitor it. There’s a core integration for postfix, and a community integration for sendmail, but not for exim. It’s also been requested before at #1721. 

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

It’s been a bit of a mash-up of the postfix and sendmail integrations.

### CODE REVIEW

**Comment by kayayarai** [comment]
Opened DOCS-3565 to track editorial review by docs team

File: exim/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][3] for guidance on applying these instructions.
+
+### Installation
+
+To install the exim check on your host:
+
+
+1. Install the [developer toolkit]
+(https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit)
+ on any machine.
```
Comment (buraizu) [inline exim/README.md:18]:
```suggestion
1. Install the [developer toolkit][10] on any machine.
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Small correction for a link

File: exim/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+1. Install the [developer toolkit][10] on any machine.
+
+2. Run `ddev release build exim` to build the package.
+
+3. [Download the Datadog Agent][2].
+
+4. Upload the build artifact to any host with an Agent and
+ run `datadog-agent integration install -w
+ path/to/exim/dist/<ARTIFACT_NAME>.whl`.
```
Comment (yzhan289) [inline exim/README.md:24]: Once this integration is released, there's no need to download the developer toolkit. Customers will just be able to install the integration with instructions [from here](https://docs.datadoghq.com/agent/guide/use-community-integrations/?tab=agentv721v621).

File: exim/datadog_checks/exim/check.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import re
+from collections import namedtuple
+from typing import Any
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.utils.subprocess_output import get_subprocess_output
+
+# from datadog_checks.base.utils.db import QueryManager
+# from requests.exceptions import ConnectionError, HTTPError, InvalidURL, Timeout
+# from json import JSONDecodeError
```
Comment (yzhan289) [inline exim/datadog_checks/exim/check.py:10]: Should this be removed?

File: exim/manifest.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{
+  "display_name": "exim",
+  "maintainer": "a",
```
Comment (yzhan289) [inline exim/manifest.json:3]: Can a real maintainer contact info be put here?

**Comment by JeanFred** [comment]
Thanks for the review @buraizu , I believe I addressed the comments.

File: exim/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+This check monitors [exim][1] through the Datadog Agent.
+
+## Setup
+
+Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][3] for guidance on applying these instructions.
+
+### Installation
+
+The exim check is not included in the [Datadog Agent][2] package, so you need to install it.
```
Comment (hithwen) [inline exim/README.md:13]:
```suggestion
### Installation

For Agent v7.21+ / v6.21+, follow the instructions below to install the Exim check on your host. See [Use Community Integrations][10] to install with the Docker Agent or earlier versions of the Agent.

1. Run the following command to install the Agent integration:

   ```shell
   datadog-agent integration install -t datadog-exim==<INTEGRATION_VERSION>
   ```

2. Configure your integration similar to core [integrations][11].
```

File: exim/README.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+[1]: https://www.exim.org/
+[2]: https://app.datadoghq.com/account/settings#agent
+[3]: https://docs.datadoghq.com/agent/kubernetes/integrations/
+[4]: https://github.com/DataDog/integrations-core/blob/master/check/datadog_checks/check/data/conf.yaml.example
+[5]: https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent
+[6]: https://docs.datadoghq.com/agent/guide/agent-commands/#agent-status-and-information
+[7]: https://github.com/DataDog/integrations-core/blob/master/check/metadata.csv
+[8]: https://github.com/DataDog/integrations-core/blob/master/check/assets/service_checks.json
+[9]: https://docs.datadoghq.com/help/
+[10]: https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit
```
Comment (hithwen) [inline exim/README.md:53]:
```suggestion
[10]: https://docs.datadoghq.com/agent/guide/use-community-integrations/
[11]: https://docs.datadoghq.com/agent/guide/use-community-integrations/
```

File: exim/datadog_checks/exim/check.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'exim'
+
+    SERVICE_CHECK_NAME = 'returns.output'
+
+    def __init__(self, name, init_config, instances):
+        super(EximCheck, self).__init__(name, init_config, instances)
+
+    def check(self, instance):
```
Comment (hithwen) [inline exim/datadog_checks/exim/check.py:19]:
```suggestion
    def check(self, _):
```
this parameter is deprecated. You can access the instance at any point with `self.instance`

File: exim/manifest.json [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  "guid": "04552816-c3e6-4a0a-90a0-bd751633984d",
+  "support": "contrib",
+  "supported_os": [
+    "linux",
+    "mac_os",
+    "windows"
+  ],
+  "public_title": "exim",
+  "categories": [
+    ""
```
Comment (hithwen) [inline exim/manifest.json:19]: These are the categories on this page, see if any one fits https://docs.datadoghq.com/integrations/

**Review by hithwen — CHANGES_REQUESTED** [review]

File: exim/manifest.json (hunk 8 above)
Comment (JeanFred) [reply exim/manifest.json:19]: Picked `Collaboration`, like sendmail and postfix

**Comment by JeanFred** [comment]
I believe I addressed all the comments. Looks like the build failures come from a broken master branch.

File: exim/README.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+For Agent v7.21+ / v6.21+, follow the instructions below to install the exim check on your host. See [Use Community Integrations][3] to install with the Docker Agent or earlier versions of the Agent.
+
+1. Run the following command to install the Agent integration:
+
+   ```shell
+   datadog-agent integration install -t datadog-exim==<INTEGRATION_VERSION>
+   ```
+
+2. Configure your integration similar to core [integrations][4].
```
Comment (buraizu) [inline exim/README.md:21]:
```suggestion
2. Configure your integration similar to core [integrations][2].
```

File: exim/README.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+See [service_checks.json][9] for a list of service checks provided by this integration.
+
+## Troubleshooting
+
+Need help? Contact [Datadog support][10].
+
+
+[1]: https://www.exim.org/
+[3]: https://app.datadoghq.com/account/settings#agent
```
Comment (buraizu) [inline exim/README.md:53]:
```suggestion
[2]: https://docs.datadoghq.com/getting_started/integrations/
[3]: https://docs.datadoghq.com/agent/guide/use-community-integrations/
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks, this looks good, just a couple of links not opening the expected pages

**Comment by FlorentClarret** [comment]
Similar to https://github.com/DataDog/integrations-extras/pull/1311#issuecomment-1198358336, I fixed the models files :)

**Comment by JeanFred** [comment]
> Thanks, this looks good, just a couple of links not opening the expected pages

Thanks @buraizu , PR amended.

File: exim/README.md [hunk 11]
```diff
@@ -0,0 +1,61 @@
+# Agent Check: exim
+
+## Overview
+
+This check monitors [exim][1] through the Datadog Agent.
```
Comment (buraizu) [inline exim/README.md:5]:
```suggestion
This check monitors [Exim][1] through the Datadog Agent.
```

File: exim/manifest.json [hunk 12]
```diff
@@ -0,0 +1,35 @@
+{
+  "display_name": "exim",
+  "maintainer": "JeanFred1@gmail.com",
+  "manifest_version": "1.0.0",
+  "name": "exim",
+  "metric_prefix": "exim.",
+  "metric_to_check": "exim.queue.count",
+  "creates_events": false,
+  "short_description": "exim integration to monitor mail queues",
```
Comment (buraizu) [inline exim/manifest.json:9]:
```suggestion
  "short_description": "Exim integration to monitor mail queues",
```

File: exim/manifest.json [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+  "guid": "04552816-c3e6-4a0a-90a0-bd751633984d",
+  "support": "contrib",
+  "supported_os": [
+    "linux",
+    "mac_os",
+    "windows"
+  ],
+  "public_title": "exim",
```
Comment (buraizu) [inline exim/manifest.json:17]:
```suggestion
  "public_title": "Exim",
```

File: exim/README.md [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+
+## Data Collected
+
+### Metrics
+
+See [metadata.csv][8] for a list of metrics provided by this integration.
+
+### Events
+
+The exim integration does not include any events.
```
Comment (buraizu) [inline exim/README.md:41]:
```suggestion
The Exim integration does not include any events.
```

**Review by buraizu — APPROVED** [review]
Thanks, just one more minor edit -- I noticed that `Exim` is capitalized in their own docs:
https://www.exim.org/
Otherwise LGTM!

**Comment by JeanFred** [comment]
> Thanks, just one more minor edit -- I noticed that `Exim` is capitalized in their own docs: https://www.exim.org/ Otherwise LGTM!

Now capitalized

File: exim/datadog_checks/exim/__about__.py [hunk 15]
```diff
@@ -0,0 +1 @@
+__version__ = '0.0.1'
```
Comment (fanny-jiang) [inline exim/datadog_checks/exim/__about__.py:1]:
```suggestion
__version__ = '1.0.0'
```

**Review by fanny-jiang — COMMENTED** [review]
Looks great! I have just one comment to update the check version in the `__about__.py` file.

File: exim/datadog_checks/exim/__about__.py (hunk 15 above)
Comment (JeanFred) [reply exim/datadog_checks/exim/__about__.py]: Done

**Review by fanny-jiang — APPROVED** [review]

**Comment by fanny-jiang** [comment]
Hi @JeanFred, I've merged the PR. Should I go ahead and push the integration tile to prod?

**Comment by JeanFred** [comment]
Forgot to mention that this resolves #1721

**Pull request merged**